dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCClient.scala

```diff
@@ -0,0 +1,1056 @@
+
+    val fundingPSBT = remoteSigs.zipWithIndex.foldLeft(
+      PSBT.fromUnsignedTx(createUnsignedFundingTransaction)) {
+      case (psbt, (sig, index)) =>
+        psbt
+          .addWitnessUTXOToInput(remoteFundingInputs(index)._2,
+                                 index + remoteTweak)
+          .addSignature(sig, index + remoteTweak)
+    }
+
+    val signedFundingPSBTF =
+      fundingUtxos.zipWithIndex.foldLeft(Future.successful(fundingPSBT)) {
+        case (psbtF, (utxo, index)) =>
+          psbtF.flatMap { psbt =>
+            psbt
+              .addWitnessUTXOToInput(output = utxo.output, index + localTweak)
+              .addScriptWitnessToInput(
+                scriptWitness = utxo.scriptWitnessOpt.get,
+                index + localTweak)
+              .sign(index + localTweak, utxo.signers.head)
+          }
+      }
+
+    signedFundingPSBTF.flatMap { signedFundingPSBT =>
+      val txT =
+        signedFundingPSBT.finalizePSBT.flatMap(_.extractTransactionAndValidate)
+      Future.fromTry(txT)
+    }
+  }
+
+  /** Constructs CET given sig*G, the funding tx's UTXOSpendingInfo and payouts */
+  def createCET(
+      sigPubKey: ECPublicKey,
+      remoteSig: PartialSignature,
+      payout: CurrencyUnit,
+      remotePayout: CurrencyUnit): (Future[Transaction], P2WSHWitnessV0) = {
+    val (cetPrivKey, cetRemotePubKey) = if (sigPubKey == sigPubKeyWin) {
+      (cetWinPrivKey, cetRemoteWinPubKey)
+    } else {
+      (cetLosePrivKey, cetRemoteLosePubKey)
+    }
+
+    val toLocalPrivKey =
+      cetPrivKey.deriveChildPrivKey(UInt32.zero).key
+    val remoteToLocalPubKey =
+      cetRemotePubKey.deriveChildPubKey(UInt32.one).get.key
+    val toRemotePubKey = cetRemotePubKey.deriveChildPubKey(UInt32(2)).get.key
+
+    val pubKeyBytes = NativeSecp256k1.pubKeyTweakAdd(
+      sigPubKey.bytes.toArray,
+      toLocalPrivKey.bytes.toArray,
+      true)
+    val pubKey = ECPublicKey.fromBytes(ByteVector(pubKeyBytes))
+
+    val toLocalSPK = P2PKWithTimeoutScriptPubKey(
+      pubKey = pubKey,
+      lockTime = ScriptNumber(timeouts.penaltyTimeout),
+      timeoutPubKey = remoteToLocalPubKey
+    )
+
+    val toLocal: TransactionOutput =
+      TransactionOutput(payout + toLocalClosingFee,
+                        P2WSHWitnessSPKV0(toLocalSPK))
+    val toRemote: TransactionOutput =
+      TransactionOutput(remotePayout, P2WPKHWitnessSPKV0(toRemotePubKey))
+
+    val outputs: Vector[TransactionOutput] = Vector(toLocal, toRemote)
+
+    val fundingTx = createUnsignedFundingTransaction
+    val fundingTxid = fundingTx.txId
+    val fundingInput = TransactionInput(
+      TransactionOutPoint(fundingTxid, UInt32.zero),
+      EmptyScriptSignature,
+      sequence)
+
+    val psbt = PSBT.fromUnsignedTx(
+      BaseTransaction(TransactionConstants.validLockVersion,
+                      Vector(fundingInput),
+                      outputs,
+                      timeouts.contractMaturity.toUInt32)
+    )
+
+    val readyToSignPSBT = psbt
```

Comment:
important change: one signature is given as input to local CETs so PSBTs are used